src/support/tx.js

```diff
@@ -400,44 +390,64 @@ tx.createPlatformCoinTransferTx = async function({
   txOptions = {}
 }) {
   // Merging txOptions
-  const updatedTxOptions = _.merge({}, { gasPrice: TX_GAS_PRICE, chainId: TX_CHAIN_ID }, txOptions);
-
-  // Generating nonce if required
-  let noncePromise = updatedTxOptions.nonce ? Promise.resolve(updatedTxOptions.nonce) : undefined;
-  if (!noncePromise) {
-    noncePromise = assignedNonce ? nonceManager.getNonceForAccount(fromAddress) : Promise.resolve();
+  const updatedTxOptions = _.defaults({}, txOptions, { gasPrice: TX_GAS_PRICE, chainId: TX_CHAIN_ID });
```

Comment:
@dankelleher 2 questions: 
1. What do think about this change?  Functionally it is the same but I believe `defaults` is more appropriate semantically. If you agree, I will change in other places as well.
2. `updatedTxOptions` Do you think we need to create this new object just to avoid `txOptions` argument mutation?  If so, can we rename the argument and use `txOptions` for merged one as it is quite a long name and it is referred multiple times.

Reply:
I agree with both of those points